### DISCUSSION TITLE
New install fails to start if apache installed - but need apache

### DISCUSSION THREAD

**Discussion by rogercreagh** [AzuraCast/AzuraCast#6885] [General]
I have made a fresh install of azuracast which I intend to run on a subdomain on the server. However the server hosts 3 domains and various subdomains already all using apache2 and letsencrypt for ssl 

Trying to access http:.//yyy.zzz.aaa:8080 sfter install to configure azuracast just results in a spinning wheel of doom and a timeout.

Surely azuracast doesn't require its own dedicated server? It is not feasible to switch all the existing sites to nginx from apache, and the parent domain for the subdomain I want azuracast on needs to stay on apache - though obviously the subdomain could be on ngnix.

Is these even feasible - if so how? the docs are silent on the subject.

**Comment by BusterNeece** [comment]
@rogercreagh Did you set AzuraCast's HTTP port to be 8080? By default, it listens on port 80 directly, as it would if it were the only thing installed on the server, but you can change that via `./docker.sh setup-ports` or by modifying the settings during the install/update process.

**Reply by rogercreagh** [reply]
yes I believe I did set 8080 and also 8443 for https.  Just checked and they are the defaults given by `./docker.sh setup-ports`
http://x.y.z.a:80 gives the apache default page (I actually have phpmyadmin installed using the apache default.config site and that is still working correctly)
The install followed the instructions in manual page. I have tried uninstalling it and reinstalling it but still the same spinning wheel of doom.

**Reply by rogercreagh** [reply]
`sudo lsof -i :8080` gives the following response:
```
COMMAND    PID USER   FD   TYPE DEVICE SIZE/OFF NODE NAME
docker-pr 4215 root    4u  IPv4  48461      0t0  TCP *:http-alt (LISTEN)
docker-pr 4221 root    4u  IPv6  49159      0t0  TCP *:http-alt (LISTEN)
```
and for :80 it is all apache2

**Comment by rogercreagh** [comment]
@BusterNeece or anyone - any other suggestions? The ports seem to be set correctly. The docker container seems to be in place.
The apache2 sites-enabled config for the default `000-default.conf` only has an entry for  `<VirtualHost *:80>` , no other apache site is using 8080 as far as I can tell. 

I have tried  changing the port used by azuracast to 8081 but the result is exactly the same.

Is there anything else that could be using 8080 - as I said above `lsof -i :8080` only shows docker as listening on port 8080

Is it in fact impossible for azuracast in docker to co-exist with apache on the same server?

**Reply by gAlleb** [reply]
1. Try `./docker.sh setup-ports` and change to something other to check for sure (e.g. 10080 and 10443)

2. Whats your OS? Check your you firewall settings! (e.g. `sudo ufw status` on Ubuntu). Ports must be open. I think problem might be here. Sometimes people forget.

>Is it in fact impossible for azuracast in docker to co-exist with apache on the same server?

Sure it is possible. If nothing helps you can always bind subdomain to dedicated AzuraCast server.

**Comment by rogercreagh** [comment]
Ahha. Brilliant clue @gAlleb  - many thanks. I knew 8080 was not blocked on the server, but you prompted me to check on the VPS provider and sure enough the default firewall there was only allowing 80 and 443. Doh!
Now I'm on the way

**Reply by gAlleb** [reply]
Good to hear! Don't forget: some other ports need to be opened too (e.g. Icecast Broadcasting Ports). Or in AzuraCast Settings you can always choose "Use Web Proxy for Radio" to route all radio ports through the web port.

**Comment by rogercreagh** [comment]
All well UNTIL I Got to the end of the setup routine and now it is all inaccessible. It gives anerror on both the ip address and the subdomain.

`Connection is insecure (ipaddr) sent an invalid response. ERR_SSL_PROTOCOL_ERROR` 

I set it to be on (radio.mydomain.com). Letsencrypt only works with domain names so why is the IP addrr:8080 now inaccessible? Surely it shouldn't have set that to SSL only. The radio.mydomain.com also gives the same error. That had a letsencrypt cert installed, but I guess the docker web process can't find it as it was installed on apache?

Furthermore the `docker.sh` app has disappeared from `/var/azuracast/` so I can't do anything there to fix it.

This is all a bit of a nightmare to set up, I can only guess I made some mistake in answering one of the setup questions and I am going to have to try and kill the whole thing and start again. I wish there was some better guidance during the setup proces

Is there an apache version of azuracast?

**Reply by gAlleb** [reply]
During setup you've been asked for 2 port adresses. One for http and one for https. Since you've installed certs, instance should be available. Which port have you chosen for https? 8443? Then you can access by `https://radio.mydomain.com:[PORT_FOR_HTTPS]`

> why is the IP addrr:8080 now inaccessible?

It could be unaccessible because of browser cache and setup.

> The radio.mydomain.com also gives the same error. but I guess the docker web process can't find it as it was installed on apache?

If you have installed Azuracast on `https://radio.mydomain.com:[PORT_FOR_HTTPS]` then it has nothing to do with `https://radio.mydomain.com:433` or `http://radio.mydomain.com:80`. Docker just forwards ports from inside the container  to the ports you've selected during installattion proccess. 

If you are using `radio.mydomain.com` on default ports with apache - you should look into apache config. But since you've installed Azuracast on custom (not 80 and 443) ports I'm not sure I understand the question.

**Reply by rogercreagh** [reply]
I wonder if my mistake/misunderstanding was assuming that azurcast in docker could use the letsencrypt cert which was installed on the apache version of the radio.mydomain.com site.

Since both ip addr and domain name are giving the same error on both 8080 and 8443 I'm wondering if by simply checking use https during the setup azuracast is trying to forward 8080 to ssl 8443 with no cert available. I should have told it to request a fresh cert?

Is there any way to fix this without starting afresh. I seem to have locked myself out and as I said the `docker.sh` script seems to have been deleted during setup so I now can't change anything from outside the container?

**Reply by gAlleb** [reply]
> I wonder if my mistake/misunderstanding was assuming that azurcast in docker could use the letsencrypt cert which was installed on the apache version of the radio.mydomain.com site.

No, it can't out of the box. Unless you reverse proxy it through apache, which is not your case. 
But you can create a docker-compose.override.yml with certs from your host.

```
services:
  web:
    volumes:
      - /path/on/host/to/ssl.crt:/var/azuracast/acme/ssl.crt:ro
      - /path/on/host/to/ssl.key:/var/azuracast/acme/ssl.key:ro
```
Please see https://www.azuracast.com/docs/administration/ssl-and-lets-encrypt/
> Since both ip addr and domain name are giving the same error on both 8080 and 8443 I'm wondering if by simply checking use https during the setup azuracast is trying to forward 8080 to ssl 8443 with no cert available. I should have told it to request a fresh cert?

When you are installing Azuracast - you select ports (8080 and 8433). 
But you must take into consideration that in order to install certs via Azuracast:

```
AzuraCast’s web server must be served on the default ports, 80 for HTTP and 443 for HTTPS. 
By default, AzuraCast is already set up this way, but if you’ve modified the ports to serve the site on a secondary port, 
you must switch the ports back to the defaults when setting up LetsEncrypt and when performing renewals.

```
1. You can stop apache service during install of Azuracast. 
2. Intsall Azuracast on default ports (80 and 443). 
3. Get the certs via web-interface http://radio.mydomain.com:80
4. Then change ports back to 8080 and 8443 via ./docker.sh setup-ports. Restart docker container. 
5. Restart apache. 
6. After you've installed certs - you can access Azuracast web by https (https://radio.mydomain.com:8443).

> Is there any way to fix this without starting afresh. I seem to have locked myself out and as I said the docker.sh script seems to have been deleted during setup so I now can't change anything from outside the container?

It could not have been deleted by itself. But yeah - its a bummer. Maybe starting out from a clean sheet is a best option.

**Reply by rogercreagh** [reply]
Excellent advice, thank you very much. I finally got it working after a couple of uninstalls and minor consequential problems.

I then discovered that something during one of the uninstalls had manged to uninstall php8.1-mysqli as well - which of course killed all of the other websites using mariadb through php  - that took some stress to sort out.

So I managed to get a test station playing ok, albeit with all other sites on the server dead. Whilst trying to work out what had gone wrong I decided I'd have to uninstall azuracast and docker again to sort it out.

The upshot of all this is that I realise azuracast really does need to run on its own server (which is what the live one ultimately will be doing) - I've just been trying to build a test setup to try things out before replicating them onto the live site, and there are just too many compromises sharing with Apache on a limited capacity VPS. So I'll be having to set up a dedicated test server instead. 

So many thanks to you for your help in getting this far. I expect I'll have more questions once I've sorted out a server and got things running, but at least I know it does seem to work ok for what I need.
